UI/cmake/os-linux.cmake

```diff
@@ -8,6 +8,11 @@ if(TARGET OBS::python)
   target_link_options(obs-studio PRIVATE LINKER:-no-as-needed)
 endif()
 
+# USE_XDG is only used once for a workaround with ealier versions of Qt
+if(Qt6_VERSION VERSION_LESS 6.3)
+  target_compile_definitions(obs-studio PRIVATE USE_XDG)
+endif()
```

Comment:
As far as I can tell, USE_XDG is already defined for all UI files that include `UI/ui-config.h` if it is defined via CMake:
https://github.com/obsproject/obs-studio/blob/52660beae2af3b5cbb792823248d52d628dcfdcb/UI/ui-config.h.in#L3
https://github.com/obsproject/obs-studio/blob/52660beae2af3b5cbb792823248d52d628dcfdcb/UI/obs-app.cpp#L79

If USE_XDG is set via CMake, the former would become:
```cpp
#define USE_XDG <value>
```

We just no longer define it as an option by default.

Reply:
The CMake option has been dropped in version 3, USE_XDG is only defined to libobs on Linux.

But for the UI, since it is only used for one thing, I preferred softly deprecate its use.
Once we drop support for Qt 6.2, USE_XDG in UI and the override can be removed.

Reply:
The only thing that was dropped in CMake 3 was that we no longer expose the option by default. You can still set `USE_XDG` in CMake 3 and it will be configured and defined. See:
https://github.com/obsproject/obs-studio/blob/52660beae2af3b5cbb792823248d52d628dcfdcb/UI/CMakeLists.txt#L55

![image](https://github.com/obsproject/obs-studio/assets/624931/27594c96-b185-41cb-bc1c-3702422be03f)

Reply:
I will not add back the option, this CMake 2 only thing.

I will just remove the Qt version check in CMake, and put it as a common compilation define like in libobs.

Reply:
I wasn't suggesting to add back the option. I'm pointing out that you can, with CMake 3, simply define `USE_XDG=true`, and the `#cmakedefine USE_XDG` will correctly transform into `#define USE_XDG` such that `#if defined(USE_XDG)` is true. There is no need to add it as an expliciti CMake `option()`.